core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+var _ types.ReportingPlugin = &promPlugin{}
+
+var (
+	promQuery = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_query_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Query() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "plugin"},
+	)
+	promObservation = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_observation_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Observation() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Report() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldAcceptFinalizedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_accept_finalized_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's ShouldAcceptFinalizedReport() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldTransmitAcceptedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_transmit_accepted_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's ShouldTransmitAcceptedReport() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promClose = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_close_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Close() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"chainType", "chainID", "pluginName"},
+	)
+)
+
+// promPlugin wraps another OCR2 reporting plugin and provides standardized prometheus metrics
+// for each of the OCR2 phases (Query, Observation, Report, ShouldAcceptFinalizedReport,
+// ShouldTransmitAcceptedReport, and Close).
+type promPlugin struct {
+	wrapped    types.ReportingPlugin
+	pluginName string
+	evmChainID *big.Int
+}
+
+func New(plugin types.ReportingPlugin, pluginName string, evmChainID *big.Int) types.ReportingPlugin {
+	return &promPlugin{
+		wrapped:    plugin,
+		pluginName: pluginName,
+		evmChainID: evmChainID,
+	}
+}
+
+func (p *promPlugin) Query(ctx context.Context, timestamp types.ReportTimestamp) (types.Query, error) {
+	start := time.Now().UTC()
+	defer func() {
+		duration := float64(time.Now().UTC().Sub(start))
+		promQuery.WithLabelValues(p.evmChainID.String(), p.pluginName).Observe(duration)
```

Comment:
Once you update the labels above you should specify as much labels here, in the same order. In this case it'll be `WithLabelValues(p.chainType, p.evmChainID.String(), p.pluginName).Observe(duration)`